File: Lecture 2/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string sentence);
int count_words(string sentence);
int count_sentences(string sentences);

int main(void)
{
    string sentence = get_string("Text: ");

    int letters = count_letters(sentence), words = count_words(sentence), sentences = count_sentences(sentence);

    // printf("%i letters\n", letters);

    // printf("%i words\n", words);

    // printf("%i sentences\n", sentences);

    double index = ((0.0588 * (((float) letters / words) * 100)) - (0.296 * (((float) sentences / words) * 100))) - 15.8;

    // printf("%f\n", index);

    int result = round(index);

    if (result < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (result >= 16)
    {
        //loool what am I doind ?
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", result);
    }


}

// more comments)
int count_letters(string sentence)
{
    int counter = 0;
    for (int i = 0, n = strlen(sentence); i < n; i++)
    {
        //emioglu
        if ((sentence[i] >= 65 && sentence[i] <= 90) || (sentence[i] >= 97 && sentence[i] <= 122))
        {
            counter++;
        }
    }

    return counter;
}

// more and moreeeee
int count_words(string sentence)
{
    int counter;

    if (sentence[0] == 32)
    {
        counter = 0;
    }
    else
    {
        counter = 1;
    }
    //simple for loop)
    for (int i = 0, n = strlen(sentence); i <= n; i++)
    {

        if (sentence[i] == 32 && isalpha(sentence[i + 1]))
        {
            counter++;
        }

    }

    return counter;

}

//mystyle is the best)
int count_sentences(string sentences)
{
    int counter = 0, i = 0;
    while (sentences[i] != '\0')
    {
        //comment here too
        if (sentences[i] == 33 || sentences[i] == 46 || sentences[i] == 63)
        {
            counter++;
        }

        i++;
    }
//another comment
    return counter;
}

File: Lecture 6/sentimental-readability/readability.py
import cs50



def main():
    my_text = cs50.get_string("Text: ")

    sentences = get_sentences(my_text)

    chars = get_char(my_text)

    words = len(my_text.split())

    L = chars / words * 100

    S = sentences / words * 100

    index = round(0.0588 * L - 0.296 * S - 15.8)


    if index > 15:
        print("Grade 16+")
    elif index < 2:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


def get_sentences(txt):
    return txt.count(".") + txt.count("?") + txt.count("!")


def get_char(txt):

    characters = 0

    for char in txt:
        if char.isalpha():
            characters += 1

    return characters

if __name__ == "__main__":
    main()